src/Microsoft.PowerShell.PSReadLine/ReadLine.cs

```diff
@@ -888,33 +888,24 @@ public static void InvokePrompt(ConsoleKeyInfo? key = null, object arg = null)
         /// </summary>
         public static string GetPrompt()
         {
-            var runspaceIsRemote = _singleton._mockableMethods.RunspaceIsRemote(_singleton._runspace);
-            System.Management.Automation.PowerShell ps;
-            if (!runspaceIsRemote)
+            var command = new PSCommand();
+            command.AddCommand("prompt");
+            var results = HostUtilities.InvokeOnRunspace(command, _singleton._runspace);
```

Comment:
I cannot implement InvokeOnRunspace() in PSReadline module using PowerShell V3 APIs so I will need to add a fix here, but I will leave in the new PowerShell engine InvokeOnRunspace() method since it will be useful to host writers in the future and help them avoid this problem.

---
In reply to: [78211320](https://github.com/PowerShell/PowerShell/pull/2213#discussion_r78211320) [](ancestors = 78211320)